setup.py

```diff
@@ -59,17 +60,29 @@ def configuration(parent_package='', top_path=None):
 
 
 if __name__ == "__main__":
-    # purposely fail if numpy is not available
-    # other dependecies will be resolved by pip (install_requires)
     try:
         from numpy.distutils.core import setup
+        extra = {'configuration': configuration}
+        # do not risk updating numpy
+        INSTALL_REQUIRES = [r for r in INSTALL_REQUIRES if 'numpy' not in r]
     except ImportError:
-        print('To install scikit-image from source, you will need numpy.\n' +
-              'Install numpy with pip:\n' +
-              'pip install numpy\n'
-              'Or use your operating system package manager. For more\n' +
-              'details, see http://scikit-image.org/docs/stable/install.html')
-        sys.exit(1)
+        if len(sys.argv) >= 2 and ('--help' in sys.argv[1:] or
+                                   sys.argv[1] in ('--help-commands',
+                                                   'egg_info', '--version',
+                                                   'clean')):
+            # For these actions, NumPy is not required.
+            #
+            # They are required to succeed without Numpy for example when
+            # pip is used to install scikit-image when Numpy is not yet
+            # present in the system.
+            pass
+        else:
+            print('To install scikit-image from source, you will need numpy.\n' +
+                  'Install numpy with pip:\n' +
```

Comment:
Implicit string concatenation - no pluses needed!

I'd consider doubling the newlines on either side of the actual pip call.